src/biblebot/auth.py

```
@@ -0,0 +1,250 @@
+        logger.debug("E2EE dependencies not found, proceeding without encryption for login.")
+
+    client = AsyncClient(hs, user, config=AsyncClientConfig(store_sync_tokens=True, encryption_enabled=e2ee_available))
+
+    # Attempt server discovery to normalize homeserver URL
+    hs = await discover_homeserver(client, hs)
+    client.homeserver = hs
+
+    logger.info(f"Logging in to {hs} as {user}")
+    try:
+        resp = await client.login(password=pwd, device_name="biblebot")
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)